sdk/nodejs/Makefile

```diff
@@ -32,6 +32,7 @@ build_package:: ensure
 	node ../../scripts/reversion.js bin/package.json ${VERSION}
 	node ../../scripts/reversion.js bin/version.js ${VERSION}
 	cp -R proto/. bin/proto/
+	cp -R tests/mockpackage bin/tests/
```

Comment:
This means we end up distributing mockpackage as part of our distribution. I wasn’t clear on exactly why mockpackage needed to be included as a dependency originally, but it seems somewhat unfortunate to have to actually ship it. Is there a way we can avoid the need for the mockpackage dependency in the first place?

Reply:
I added mockpackage to write a test to check that function serialization properly works with a package that specifies exports. I could remove the test and the package.

I could also add additional scripting to the makefile to only add and remove the mockpackage dependency during testing.

There was a solution in https://github.com/yarnpkg/yarn/issues/3630#issuecomment-641959126 and it involves a one-liner that removes devDependencies out of our package.json so it doesn't have to be copied over. I was initially using it, but figured it might be less favorable to have long awk one-liners editing files.

Reply:
I have this implementation that strips the devDependencies by renaming the field.

https://github.com/pulumi/pulumi/blob/dixler/9019/fix-make-install-strip/sdk/nodejs/Makefile#L31-L32

Reply:
Using other strategy to discard devDependencies from `package.json`.